openstack/resource_openstack_lb_l7policy_v2.go

```diff
@@ -129,120 +142,92 @@ func resourceL7policyV2Create(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[DEBUG] Create Options: %#v", createOpts)
 
 	timeout := d.Timeout(schema.TimeoutCreate)
-	listenerID := createOpts.ListenerID
-
-	if v, ok := d.GetOk("redirect_url"); ok {
-		redirectURL = v.(string)
-		createOpts.RedirectURL = redirectURL
-	}
-
-	// Make sure the associated pool is active before proceeding
-	if v, ok := d.GetOk("redirect_pool_id"); ok {
-		redirectPoolID = v.(string)
-		createOpts.RedirectPoolID = redirectPoolID
 
-		err = waitForLBV2viaPool(lbClient, redirectPoolID, "ACTIVE", []string{"PENDING_CREATE", "PENDING_UPDATE"}, timeout)
+	// Make sure the associated pool is active before proceeding.
+	if redirectPoolID != "" {
+		err = waitForLBV2Pool(lbClient, redirectPoolID, "ACTIVE", lbV2PoolPendingStatuses, timeout)
```

Comment:
The `waitForLBV2Pool` function is used only to detect the `DELETED` status. We have to use `waitForLBV2viaPool` for neutron lbaasv2 extension. Will revert this commit in my PR.